### PR TITLE
Fix takeFirst handler to asyncTakeFirst

### PR DESCRIPTION
**Pull request by ditkachuk**

### CODE REVIEW

File: src/utils/saga.js [hunk 1]
```diff
@@ -1,23 +1,32 @@
...
+
+export function* asyncTakeFirst(pattern, saga, ...args) {
     const task = yield fork(function*() {
         let firstTasks = [];
         /* eslint-disable no-loop-func */
         while (true) {
             const action = yield take(pattern);
-            const params = storedBy(action);
+            const params = action.params || noParamsKey;
+            console.log(action, params);
```
Comment (megazoll) [inline src/utils/saga.js]: remove

File: src/utils/saga.js [hunk 2]
```diff
@@ -28,6 +37,7 @@ export function* takeFirst(pattern, storedBy, saga, ...args) {
     return task;
 }
 
+
```
Comment (megazoll) [inline src/utils/saga.js]: remove

File: src/utils/saga.js [hunk 3]
```diff
@@ -4,21 +4,21 @@ import contains from 'ramda/src/contains';
 import not from 'ramda/src/not';
 import without from 'ramda/src/without';
 
-export function* takeFirst(pattern, storedBy, saga, ...args) {
+export function* asyncTakeFirst(pattern, saga, ...args) {
```
Comment (megazoll) [inline src/utils/saga.js]: create `create` function with optional `storedBy` argument

Comment (ditkachuk) [reply src/utils/saga.js]: fixed

File: package.json [hunk 4]
```diff
@@ -48,7 +48,7 @@
     "react-dom": ">=15.4.2",
     "react-redux": ">=5.0.3",
     "redux": ">=3.6.0",
-    "redux-saga": ">=0.15.3"
+    "redux-saga": ">=0.16.0"
```
Comment (megazoll) [inline package.json]: rollback

**Pull request merged**